src/webpack/CompilerEventsFacade.ts

```diff
@@ -15,16 +15,21 @@ export default class CompilerEventsFacade {
     return this._compiler.hooks.compilation.tap(
       CompilerEventsFacade.extensionName,
       (comp) => {
-        const afterOptimizeChunkAssets = (chunks) => {
-          call(comp, chunks);
+        const chunks = new Set();
+        const afterOptimizeChunkAssets = (chunksAfterOptimization) => {
+          call(comp, chunksAfterOptimization);
         };
-        /* https://github.com/webpack/webpack/blob/main/lib/Compilation.js#L772-L779
-        afterOptimizeChunkAssets = PROCESS_ASSETS_STAGE_OPTIMIZE + 1
-        */
-        const stage = Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE + 1;
-        (comp.hooks as any).processAssets.tap(
-          { name: CompilerEventsFacade.extensionName, stage },
-          afterOptimizeChunkAssets,
+        comp.hooks.processAssets.tap(
+          {
+            name: CompilerEventsFacade.extensionName,
+            stage: Compilation.PROCESS_ASSETS_STAGE_ANALYSE,
+          },
+          () => {
+            comp.chunks.forEach((chunk) => {
```

Comment:
I wonder why is this needed. Isn't `comp.chunks` already a `Set`? According to the Webpack docs it should be...